jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CachedContentFactory.java

```diff
@@ -405,7 +406,7 @@ public class CachedHttpContent implements HttpContent
 
             _lastAccessed = Instant.now();
 
-            _etag = CachedContentFactory.this._etags ? new PreEncodedHttpField(HttpHeader.ETAG, resource.getWeakETag()) : null;
+            _etag = CachedContentFactory.this._etags ? new PreEncodedHttpField(HttpHeader.ETAG, EtagUtil.calcWeakEtag(resource.getPath())) : null;
```

Comment:
This would only work with path-backed resources. Shouldn't we be capable of calculating the etag of a potential `MemoryResource`?

Reply:
There's no ETag support for MemoryResource as it exists currently.

The HttpContentFactory reorg PR (after this one) will remove MemoryResource in favor of a StaticHttpContent object fallback instead (which can hold the etag based on the URI and JAR file last timestamp)

Reply:
Fixed in this PR anyway.